input.go

```diff
@@ -1,24 +1,30 @@
 package shaping
 
 import (
+	"github.com/benoitkugler/textlayout/language"
 	"github.com/go-text/di"
 	"github.com/go-text/font"
 	"golang.org/x/image/math/fixed"
 )
 
-type Input interface {
-	// Text returns the characters to be shaped, which is `text[start:end]`.
-	//
-	// When shaping part of a larger text (e.g. a run of text from a line),
-	// instead of passing just the substring corresponding to the run, it is preferable to pass the whole
-	// line and specify the run `start` and `end` to give the shaper the full context to be able,
-	// for example, to do cross-run Arabic shaping or properly handle combining marks at start of run.
-	Text() (text []rune, start, end int)
-	// Direction returns the directionality of the text.
-	Direction() di.Direction
-	// Face returns the font face to render the text in.
-	Face() font.Face
-	// Size returns the size of the font, eg. 14.
+type Input struct {
```

Comment:
Nitpicking : maybe use a regular named field for `Direction`, `Face`, `Script`, `Language`, since there is no idea of type hierarchy (inheritance) here ?